test/clj/rems/application/test_eraser.clj

```diff
@@ -1,58 +1,235 @@
-    (let [app-id-1 (add-dummy-application! {:draft? true :date-time (time/now) :actor "alice"})
-          app-id-2 (add-dummy-application! {:date-time (time/now) :actor "alice"})
-          app-id-3 (add-dummy-application! {:date-time (time/minus (time/now) (time/days 120)) :actor "alice"})
-          expired-app-id (add-dummy-application! {:draft? true
-                                                  :date-time (time/minus (time/now) (time/days 90) (time/seconds 1))
-                                                  :actor "alice"})
-          expired-app-id-2 (add-dummy-application! {:draft? true
-                                                    :date-time (time/minus (time/now) (time/days 120))
-                                                    :actor "alice"})]
-      (let [before-apps (applications/get-all-applications "alice")]
-        (is (= #{app-id-1 app-id-2 app-id-3 expired-app-id expired-app-id-2}
-               (set (map :application/id before-apps)))))
-      (eraser/remove-expired-applications!)
-      (let [after-apps (applications/get-all-applications "alice")]
-        (is (= #{app-id-1 app-id-2 app-id-3}
-               (set (map :application/id after-apps))))))))
\ No newline at end of file
+(defn- expiration-notification-events [app-id]
+  (->> (applications/get-application-internal app-id)
+       :application/events
+       (filter (comp #{:application.event/expiration-notifications-sent}
+                     :event/type))))
+
+(defn- get-all-application-ids [user-id]
+  (->> (applications/get-all-applications user-id)
+       (map :application/id)))
+
+(defn- log*-mock [coll-atom]
+  (fn [logger level throwable message]
+    (swap! coll-atom conj {:logger logger :level level :throwable throwable :message message})))
+
+(deftest test-expire-application
+  (binding [command/*fail-on-process-manager-errors* true]
+    (test-helpers/create-user! {:userid "alice"})
+    (let [now (time/now)
+          draft (create-application! {:draft? true
+                                      :date-time now
+                                      :actor "alice"})
+          old-submitted (create-application! {:date-time (time/minus now (time/days 120))
+                                              :actor "alice"})
+          expired-draft (create-application! {:draft? true
+                                              :date-time (time/minus now (time/days 90) (time/seconds 1))
+                                              :actor "alice"})
+          outbox-emails (atom [])
+          log-messages (atom [])]
+
+      (testing "does not process applications when expirer-bot user does not exist"
+        (is (= #{draft old-submitted expired-draft} (set (get-all-application-ids "alice"))))
+        (with-redefs [env {:application-expiration {:application.state/draft {:delete-after "P90D"}}}
+                      outbox/put! (fn [email] (swap! outbox-emails conj email))
+                      log/log* (log*-mock log-messages)]
```

Comment:
Should we move this to a higher level `with-redefs` so that we don't have to repeat them in every test?